### PR TITLE
fix: handle inefficient assignments

### PR DESCRIPTION
**Pull request by steebchen**
# Description

Enables linting for inefficient assignments, and fixes all existing issues.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## What's Changed

- [x] add ineffassign linter to golangcilint
- [x] resolve or ignore issues

### CODE REVIEW

File: internal/repository/prisma/step_run.go [hunk 1]
```diff
@@ -452,22 +452,29 @@ func (s *stepRunRepository) QueueStepRun(ctx context.Context, tenantId, stepRunI
...
+		}
 
-		return err
+		return nil
 	})
 
-	err = retrier(s.l, func() error {
+	if retrierErr != nil {
+		return nil, fmt.Errorf("could not queue step run: %w", err)
+	}
```
Comment (steebchen) [inline internal/repository/prisma/step_run.go:475]: please double check @abelanger5 that this is intended.

File: internal/services/controllers/workflows/queue.go [hunk 2]
```diff
@@ -228,11 +228,11 @@ func (wc *WorkflowsControllerImpl) queueWorkflowRunJobs(ctx context.Context, wor
 		)
 
 		if err != nil {
-			err = multierror.Append(err, fmt.Errorf("could not add job run to task queue: %w", err))
+			returnErr = multierror.Append(err, fmt.Errorf("could not add job run to task queue: %w", err))
```
Comment (steebchen) [inline internal/services/controllers/workflows/queue.go:231]: this never returned the actual multierror 😅

File: internal/repository/prisma/step_run.go (hunk 1 above)
Comment (abelanger5) [reply internal/repository/prisma/step_run.go:475]: Yeah this looks right.

**Review by abelanger5 — APPROVED** [review]
Nice, these are all good catches

**Pull request merged**